packages/cli-core/src/format/format.ts

```diff
@@ -9,19 +9,38 @@ import {
   red,
   underline,
 } from 'kleur/colors';
+import { PackageManagerController } from '@aws-amplify/plugin-types';
+import { PackageManagerControllerFactory } from '../package-manager-controller/package_manager_controller_factory.js';
+import { AmplifyFault } from '@aws-amplify/platform-core';
+
+/**
+ * Exported for testing only. Should not be used directly outside of tests!
+ */
+export const packageManagerControllerContainer: {
+  packageManagerController?: PackageManagerController;
```

Comment:
I tried to make `format` into a class and inject this into the ctor but I was running into circular dependency issues and untangling that was leading to more mess. This allows us to lazily resolve a package manager controller as needed when `backendCliCommand` is called and avoid circular dependency issues on initialization.

Reply:
Might be worth adding more explanation in code comments then.

Reply:
I've pulled out the "package manager runner name" bits from the package manager controller into a more lightweight component that can be used both in the package manager controller and here. With that refactor I was able to change this to a normal class with ctor injection.